backend/services/implementations/courseService.ts

```diff
@@ -0,0 +1,121 @@
+import { getErrorMessage } from "../../utilities/errorUtils";
+import logger from "../../utilities/logger";
+import {
+  CourseRequestDTO,
+  CourseResponseDTO,
+  ICourseService,
+} from "../interfaces/ICourseService";
+import MgCourse, { Course } from "../../models/course.model";
+
+const Logger = logger(__filename);
+
+class CourseService implements ICourseService {
+  async getCourse(id: string): Promise<CourseResponseDTO> {
+    let course: Course | null;
+    try {
+      course = await MgCourse.findById(id);
+      if (!course) {
+        throw new Error(`Course id ${id} not found`);
+      }
+    } catch (error: unknown) {
+      Logger.error(`Failed to get course. Reason = ${getErrorMessage(error)}`);
+      throw error;
+    }
+
+    return {
```

Comment:
double check if need to split this or can do `return course`?

Reply:
`MgCourse.findById` returns `Course`,  not `CourseResponseDTO`. Not sure if that makes a big difference: it seems like typescript can see that they both have the same attributes. I'd save leave it for now, but it's something we can discuss at the meeting on Thursday

Reply:
Perhaps even something like `course as CourseResponseDTO` would be good